axlearn/cloud/gcp/examples/dataflow_inference_custom.py

```diff
@@ -0,0 +1,205 @@
+$ docker run -it --mount type=bind,src=$HOME/.config/gcloud,dst=/root/.config/gcloud \
+    --entrypoint /bin/bash ${DOCKER_REPO}/${DOCKER_IMAGE}:{DOCKER_TAG}
+> python3 -m axlearn.cloud.gcp.examples.dataflow_inference_custom \
+    --module=text.gpt.c4_trainer \
+    --config=fuji-7B-single  \
+    --trainer_dir='gs://.../checkpoints/step_xxx'
+
+To use axlearn CLI:
+$ axlearn gcp dataflow start \
+    --bundler_spec=dockerfile=Dockerfile \
+    --bundler_spec=repo=${DOCKER_REPO} \
+    --bundler_spec=image=${DOCKER_IMAGE} \
+    --bundler_spec=target=dataflow \
+    --bundler_spec=allow_dirty=True \
+    --dataflow_spec=runner=DataflowRunner \
+    -- "'python3 -m axlearn.cloud.gcp.examples.dataflow_inference_custom \
+        --module=text.gpt.c4_trainer \
+        --config=fuji-7B-single \
+        --trainer_dir='gs://.../checkpoints/step_xxx' \
+        '"
+
+To use GPUs for your job:
+$ axlearn gcp dataflow start \
+    --bundler_spec=dockerfile=Dockerfile \
+    --bundler_spec=repo=${DOCKER_REPO} \
+    --bundler_spec=image=${DOCKER_IMAGE} \
+    --bundler_spec=target=dataflow \
+    --bundler_spec=allow_dirty=True \
+    --dataflow_spec=runner=DataflowRunner \
+    --dataflow_spec=dataflow_service_options=\
+    "worker_accelerator=type:nvidia-l4;count:1;install-nvidia-driver" \
+    -- "'python3 -m axlearn.cloud.gcp.examples.dataflow_inference_custom \
+        --module=text.gpt.c4_trainer \
+        --config=fuji-7B-single \
+        --trainer_dir='gs://.../checkpoints/step_xxx' \
+        '"
+
+"""
+
+
+import copy
+import logging
+from typing import Any, Dict, Sequence
+
+import apache_beam as beam
+import jax
+from absl import app, flags
+from absl.flags import argparse_flags
+from apache_beam.ml.inference.base import ModelHandler, PredictionResult, RunInference
+from apache_beam.options.pipeline_options import PipelineOptions
+
+import axlearn.common.launch_trainer as trainer_utils
+from axlearn.common import input_fake
+from axlearn.common.inference import InferenceRunner, MethodRunner
+from axlearn.common.utils import NestedTensor
+
+
+class CustomModelHandler(ModelHandler[Dict, PredictionResult, Any]):
+    """Defines how to load a custom checkpoint and run inference."""
+
+    # pylint: disable-next=super-init-not-called
+    def __init__(self, flag_dict: Dict):
+        # Store absl FLAGS in a flag dictionary to avoid pickling issues
+        self._flag_dict = flag_dict
+
+    def _flag_values_from_dict(self, flag_values: Dict) -> flags.FlagValues:
+        # Avoid mutating global FLAGS.
+        fv = copy.deepcopy(flags.FLAGS)
+        for k, v in flag_values.items():
+            try:
+                fv.set_default(k, v)
+            # pylint: disable-next=protected-access
+            except flags._exceptions.UnrecognizedFlagError:
+                # Ignore unrecognized flags from other modules
+                pass
+        fv.mark_as_parsed()
+        return fv
+
+    def load_model(self) -> MethodRunner:
```

Comment:
Is this method inheriting from the parent class?

Reply:
Yes - the users simply need to implement their own way of loading the model to the workers